Userland/Libraries/LibWeb/HTML/MessagePort.cpp

```diff
@@ -320,18 +335,13 @@ ErrorOr<MessagePort::ParseDecision> MessagePort::parse_message()
     return ParseDecision::ParseNextMessage;
 }
 
-void MessagePort::read_from_socket()
+void MessagePort::read_from_transport()
 {
-    u8 buffer[4096] {};
-
-    Vector<int> fds;
-    // FIXME: What if pending bytes is > 4096? Should we loop here?
-    auto maybe_bytes = m_socket->receive_message(buffer, MSG_NOSIGNAL, fds);
-    if (maybe_bytes.is_error()) {
-        dbgln("MessagePort::read_from_socket(): Failed to receive message: {}", maybe_bytes.error());
-        return;
-    }
-    auto bytes = maybe_bytes.release_value();
```

Comment:
the "should we loop" fixme is now resolved by calling the same API that IPC does. the "perform_shutdown" callback is a bit iffy, but I found that without the global task, we ended up in the "function being destroyed while executing" invalid state of AK::Function.